contracts/src/SimpleSubmit.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IWavsServiceHandler} from "../interfaces/IWavsServiceHandler.sol";
+import {IWavsServiceManager} from "../interfaces/IWavsServiceManager.sol";
+import {ISimpleTrigger} from "../interfaces/ISimpleTrigger.sol";
+import {ISimpleSubmit} from "../interfaces/ISimpleSubmit.sol";
+
+contract SimpleSubmit is IWavsServiceHandler {
+    IWavsServiceManager private _serviceManager;
+
+    mapping(ISimpleTrigger.TriggerId => bool) validTriggers;
+    mapping(ISimpleTrigger.TriggerId => bytes) datas;
+    mapping(ISimpleTrigger.TriggerId => IWavsServiceHandler.SignatureData) signatures;
+    mapping(ISimpleTrigger.TriggerId => IWavsServiceHandler.Envelope) envelopes;
+
+    constructor(IWavsServiceManager serviceManager) {
+        _serviceManager = serviceManager;
+    }
+
+    function handleSignedEnvelope(IWavsServiceHandler.Envelope calldata envelope, IWavsServiceHandler.SignatureData calldata signatureData) external {
+        _serviceManager.validate(envelope, signatureData);
+
+        ISimpleSubmit.DataWithId memory dataWithId = abi.decode(envelope.payload, (ISimpleSubmit.DataWithId));
+
+        signatures[dataWithId.triggerId] = signatureData;
+        datas[dataWithId.triggerId] = dataWithId.data;
+        envelopes[dataWithId.triggerId] = envelope;
+        validTriggers[dataWithId.triggerId] = true;
+    }
+
+    function isValidTriggerId(ISimpleTrigger.TriggerId triggerId) external view returns (bool) {
+        return validTriggers[triggerId];
+    }
+
+    function getSignature(ISimpleTrigger.TriggerId triggerId) external view returns (IWavsServiceHandler.SignatureData memory signatureData) {
```

Comment:
This needs to be updated to newest SimpleSubmit in WAVS: https://github.com/Lay3rLabs/WAVS/blob/b55c39532f3fb477227acfdb3cea2b9db60ff093/examples/contracts/solidity/SimpleSubmit.sol#L37

Notice that the `getSignature()` and `getData()` functions are no longer separate, but rather one function call.

This isn't a superficial difference - multiple `eth_call`s can get shuffled, leading to very hard-to-detect race conditions that can show up as technically valid but stale signatures.